afkak/consumer.py

```diff
@@ -710,6 +710,13 @@ def _handle_fetch_error(self, failure):
         TODO: Possibly make this differentiate based on the failure
         """
         # The _request_d deferred has fired, clear it.
+        if failure.check(OffsetOutOfRangeError):
+            offset_request = OffsetRequest(
+                self.topic, self.partition, OFFSET_EARLIEST, 1)
```

Comment:
This makes a big assumption. This should be configurable, since in some applications losing messages (because your afkak consumer was down until the messages expired) would be a critical error.  Here you're just going merrily along, assuming that the loss of any messages between what the consumer wanted (presumably a committed offset) and what's available can be ignored.

Reply:
I do make that assumption that messages between the consumer stored offset and minimal available off set of given topic is ignorable. 

What are the parameters to make `OffsetOutOfRange` error handling configurable? `handle_ OffsetOutOfRange` (True/False), `time` (-2/-1) and `max_offsets`?

Reply:
@rthille  Do you think provide `auto.offset.reset` a feasible solution which provides certain level user involvement or user control.

Reply:
The Kafka-python approach seems reasonable, though I'd define 'earliest', 'latest', and 'raise' (explicitly), but that's just me.

Reply:
@twm  May I have your opinion as well?

Reply:
@twm ^^^

Reply:
Yes, I'd go for three options:

* EARLIEST
* LATEST
* FAIL ("fail" rather than "raise" as the failure will be reported on the Deferred returned by `start()`)

I also want to change the error handling within `Consumer` to only retry on a `BrokerResponseError` when it is retriable &mdash; see #23, which adds this metadata to the exceptions.